### ISSUE TITLE
Minetest fails to compile using OGLES

### ISSUE THREAD

**Issue by Danct12** [minetest/minetest#8810]
##### Minetest version
```
pine-pinetab:~$ DISPLAY=:0 minetest --version
Minetest 5.0.1 (Linux)
Using Irrlicht 1.9.0
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_GETTEXT=1
USE_SOUND=1
USE_CURL=1
USE_FREETYPE=1
USE_LUAJIT=1
STATIC_SHAREDIR="/usr/share/minetest"
pine-pinetab:~$ 

```

##### OS / Hardware
Operating system: postmarketOS (Alpine Linux)
CPU: Allwinner A64

GPU model: Mali 400 MP2
OpenGL version: OpenGL ES 2.0 Mesa 19.2.0-devel

##### Summary
When launch Minetest, this shows:
```
pine-pinetab:~$ DISPLAY=:0 minetest
2019-08-17 10:10:45: ERROR[Main]: Subgame specified in default_game [minetest] is invalid.
2019-08-17 10:10:47: ERROR[Main]: In thread ffff9cb9e880:                                                                              
2019-08-17 10:10:47: ERROR[Main]: /home/pmos/build/src/minetest-5.0.1/src/script/cpp_api/s_base.cpp:89: static int ScriptApiBase::luaPanic(lua_State*): A fatal error occurred: LUA PANIC: unprotected error in call to Lua API (bad light userdata pointer)                  
Aborted
```

##### Steps to reproduce
I compiled [Irrlicht GLES branch](https://github.com/zaki/irrlicht/tree/ogl-es), and Minetest. And when launch, the error above shows.

I compiled Minetest with:
```
cmake -DCUSTOM_BINDIR=/usr/bin \
		-DCMAKE_INSTALL_PREFIX=/usr \
		-DCUSTOM_DOCDIR="/usr/share/doc/$pkgname" \
		-DCUSTOM_SHAREDIR="/usr/share/$pkgname" \
		-DBUILD_CLIENT=1 \
		-DBUILD_SERVER=0 \
		-DENABLE_REDIS=0 \
		-DENABLE_CURL=1 \
		-DENABLE_GLES=1 \
		-DRUN_IN_PLACE=0 \
		..
```

**Comment by sfan5** [comment]
There have been recent changes to GLES support, please try with git master.

**Comment by Danct12** [comment]
Seems like I'm having trouble compiling the git version:
```
In file included from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/gui/guiButton.cpp:5:
/home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/gui/guiButton.h:25:8: error: multiple definition of 'enum irr::gui::EGUI_BUTTON_IMAGE_STATE'
   enum EGUI_BUTTON_IMAGE_STATE {
        ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/gui/guiButton.h:9,
                 from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/gui/guiButton.cpp:5:
/usr/include/irrlicht/IGUIButton.h:66:7: note: previous definition here
  enum EGUI_BUTTON_IMAGE_STATE
       ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/gui/guiButton.cpp:5:
/home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/gui/guiButton.h:49:59: error: redefinition of 'const c8* const irr::gui::GUIButtonImageStateNames [10]'
   const c8 *const GUIButtonImageStateNames[EGBIS_COUNT + 1] =
                                                           ^
In file included from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/gui/guiButton.h:9,
                 from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/gui/guiButton.cpp:5:
/usr/include/irrlicht/IGUIButton.h:91:18: note: 'const c8* const irr::gui::GUIButtonImageStateNames [10]' previously defined here
  const c8* const GUIButtonImageStateNames[EGBIS_COUNT+1] =
                  ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/gui/guiButton.cpp:5:
/home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/gui/guiButton.h:145:7: error: 'void GUIButton::setSprite(irr::gui::EGUI_BUTTON_STATE, irr::s32, irr::video::SColor, bool)' marked 'override', but does not override
  void setSprite(gui::EGUI_BUTTON_STATE state, s32 index, video::SColor color, bool loop) override {
       ^~~~~~~~~
make[2]: *** [src/CMakeFiles/minetest.dir/build.make:778: src/CMakeFiles/minetest.dir/gui/guiButton.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.cpp: In lambda function:
/home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.cpp:823:46: warning: 'void* irr::video::IImage::lock()' is deprecated [-Wdeprecated-declarations]
   return reinterpret_cast<u32 *>(image->lock());
                                              ^
In file included from /usr/include/irrlicht/ITexture.h:9,
                 from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.h:24,
                 from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.cpp:20:
/usr/include/irrlicht/IImage.h:178:25: note: declared here
  _IRR_DEPRECATED_ void* lock()
                         ^~~~
/home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.cpp: In lambda function:
/home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.cpp:826:17: warning: 'void irr::video::IImage::unlock()' is deprecated [-Wdeprecated-declarations]
   image->unlock();
                 ^
In file included from /usr/include/irrlicht/ITexture.h:9,
                 from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.h:24,
                 from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.cpp:20:
/usr/include/irrlicht/IImage.h:186:24: note: declared here
  _IRR_DEPRECATED_ void unlock()
                        ^~~~~~
/home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.cpp: In function 'irr::video::IImage* createInventoryCubeImage(irr::video::IImage*, irr::video::IImage*, irr::video::IImage*)':
/home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.cpp:833:53: warning: 'void* irr::video::IImage::lock()' is deprecated [-Wdeprecated-declarations]
  u32 *target = reinterpret_cast<u32 *>(result->lock());
                                                     ^
In file included from /usr/include/irrlicht/ITexture.h:9,
                 from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.h:24,
                 from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.cpp:20:
/usr/include/irrlicht/IImage.h:178:25: note: declared here
  _IRR_DEPRECATED_ void* lock()
                         ^~~~
/home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.cpp:892:17: warning: 'void irr::video::IImage::unlock()' is deprecated [-Wdeprecated-declarations]
  result->unlock();
                 ^
In file included from /usr/include/irrlicht/ITexture.h:9,
                 from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.h:24,
                 from /home/pmos/build/src/minetest-d73e458db6ab7748fcc7fd485f246971725ea48a/src/client/tile.cpp:20:
/usr/include/irrlicht/IImage.h:186:24: note: declared here
  _IRR_DEPRECATED_ void unlock()
                        ^~~~~~
make[1]: *** [CMakeFiles/Makefile2:258: src/CMakeFiles/minetest.dir/all] Error 2
```

**Comment by sfan5** [comment]
@rubenwardy you broke Irrlicht 1.9 support in d1a1c5cbf055e7eacbc4aeaa53174749d4c64d87, please fix it.

**Comment by sfan5** [comment]
The ifdef should read `#if (IRRLICHT_VERSION_MAJOR == 1) && (IRRLICHT_VERSION_MINOR <= 8)` instead and and the same(?) check needs to guard overriden `GUIButton::setSprite` too.

**Comment by sfan5** [comment]
Also `guiButton.cpp` should probably be in src/irrlicht_changes.

**Comment by rubenwardy** [comment]
Irrlicht 1.9 doesn't exist, you probably shouldn't be using a development version of Irrlicht

**Comment by sfan5** [comment]
> Is there a 1.8.4 version with ogles support?

The ogles branch (which we also use for Android) is just a development branch of Irrlicht that contains most(?) changes for 1.9 and OpenGL ES support on top of that.

**Comment by sfan5** [comment]
Fixed by ea9b241c13e9982acb993e616fa1db685ffba5cb
